ecommerce/api.py

```diff
@@ -34,23 +34,29 @@
 _REFERENCE_NUMBER_PREFIX = 'BOOTCAMP-'
 
 
-def get_total_paid(user, run_key):
+def get_total_paid(user, run_key, application_id=None):
     """
     Get total paid for a bootcamp run for a user
     Args:
         user (User):
             The purchaser of the bootcamp run
         run_key (int):
             A bootcamp run key
+        application_id (Optional[int]):
+            The id of a BootcampApplication (if the payments should only be counted if they
+            were part of the given application)
 
     Returns:
         Decimal: The total amount paid by a user for a bootcamp run
     """
-    return Line.objects.filter(
+    line_qset = Line.objects.filter(
         order__status=Order.FULFILLED,
         order__user=user,
         run_key=run_key,
-    ).aggregate(price=Sum('price'))['price'] or Decimal(0)
+    )
+    if application_id is not None:
+        line_qset.filter(order__application_id=application_id)
```

Comment:
I think you need a `line_qset = line_qset.filter(...)` here. Also, we should have a test for this case to make sure the filter is applied.